projects/blinking_led/src/main.c

```diff
@@ -0,0 +1,37 @@
+#include "interrupt.h"   // apparently soft timers needs this
+#include "log.h"         // similar to prinf but indicates the line #
+#include "soft_timer.h"  // soft timers
+#include "wait.h"        // for wait function
+
+#define COUNTER_PERIOD_MS 500  // the time between each count in ms
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void SoftTimerCallCounter(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;
+  if ((storage->counter_a + 1) % 2 != 0 || storage->counter_a == 0) {
```

Comment:
I think this would be better if you just checked and incremented counter b
It seems counter_a will increment regardless, so it would probably be better just to put it outside the if-else statement